### PR TITLE
Add hashcode implementation where equals was implemented to make the whole system more stable

### PR DESCRIPTION
**Pull request by marcocast**

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java [hunk 1]
```diff
@@ -1131,19 +1131,29 @@ protected TaskLauncherInitializer getDefaultTaskLauncherInitializer() {
...
+
     @Override
     public boolean equals(Object obj) {
-        if (obj == null) {
+        if (this == obj)
+            return true;
+        if (!super.equals(obj))
             return false;
-        }
+        if (getClass() != obj.getClass())
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java]: I think this contradicts the following check isAssignableFrom

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java [hunk 2]
```diff
@@ -570,16 +570,32 @@ public Host(InetAddress address, int nodesNumber) {
...
-                }
-                return false;
+                if (this == obj)
+                    return true;
+                if (obj == null)
+                    return false;
+                if (getClass() != obj.getClass())
+                    return false;
+                Host other = (Host) obj;
+                if (!getOuterType().equals(other.getOuterType()))
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java]: not sure that the outerType is needed as we are trying to compare two Host instances

**Comment by marcocast** [comment]
retest this please

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java (hunk 2 above)
Comment (marcocast) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java]: Done

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java (hunk 1 above)
Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java]: done

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

1. MAJOR [InternalTask.java#L85](https://github.com/ow2-proactive/scheduling/blob/01f2e48eb632806121fd698ce119a0e8872720f3/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java#L85): The Cyclomatic Complexity of this class is 201 which is greater than 200 authorized. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AClassCyclomaticComplexity)

**Pull request merged**